extensions/wikia/Forum/DiscussionsRedirectHelper.class.php

```diff
@@ -0,0 +1,77 @@
+<?
+class DiscussionsRedirectHelper {
+	/**
+	 * @return bool
+	 */
+	public static function areForumsArchivedAndDiscussionsEnabled() {
+		global $wgArchiveWikiForums, $wgEnableDiscussions;
+
+		return $wgArchiveWikiForums && $wgEnableDiscussions;
+	}
+
+	private static function discussionsUrl() {
+		return '/d';
+	}
+
+	private static function categoryUrl( $categoryId ) {
+		if ( !empty( $categoryId ) ) {
+			return '/d/f/' . $categoryId;
+		} else {
+			return self::discussionsUrl();
+		}
+	}
+	private static function postUrl( $postId ) {
+		if ( !empty( $categoryId ) ) {
+			return '/d/p/' . $postId;
+		} else {
+			return self::discussionsUrl();
+		}
+	}
+
+	private static function getDiscussionsUrl($namespace, $id) {
+		if ( $namespace == NS_WIKIA_FORUM_TOPIC_BOARD || NS_WIKIA_FORUM_BOARD ) {
+			return DiscussionsRedirectHelper::categoryUrl( $id );
+		}
+		if ( $namespace == NS_WIKIA_FORUM_BOARD_THREAD ) {
+			return DiscussionsRedirectHelper::postUrl( $id );
+		}
+		return self::discussionsUrl();
+	}
+
+	public static function redirect(
+		OutputPage $out, WikiaResponse $response
+	) {
```

Comment:
I think we can pull all this onto one without going over our max line guidelines. Can we do that?